operator_rolling_upgrades.md

```diff
@@ -0,0 +1,54 @@
+# Operator Rolling Upgrades
+Optionally allow a Infinispan CR cluster to be upgraded with zero downtime and no loss of data.
+
+## Spec
+Add a field to the CR spec that determines whether an upgrade should be `rolling` or `gracefulShutdown`. The latter is the current approach that does not allow data migration.
+
+```yaml
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: example-infinispan
+spec:
+  upgrades: rolling
+```
+
+## Procedure
+When an upgrade is possible, the operator performs the following for each Infinispan CR:
+
+1. Create a new StatefulSet for the target cluster based upon the CR spec, wait until the cluster is formed
+
+2. Retrieve all cache configurations from the cluster in the original Statefulset
+
+3. Append a remote-store configuration to each of the ^ configurations and create cache target cluster:
+
+```xml
+<remote-store xmlns="urn:infinispan:config:store:remote:12.0"
+                 cache="<cache-name>" 
+                 protocol-version="<protocol-version>"> 
+      <remote-server host="<ip-of-pod>" port="11222"/> 
+   </remote-store>
+```
+
+4. Switch clients to the target cluster.
+
+5. Update the CR spec.StatefulSetName to new Statefulset
```

Comment:
```suggestion
5. Update the CR spec.status.StatefulSetName to new Statefulset
```
We should also make StatefulSetName required and provide this filed on the reconcile ASAP